components/membership/membership-client-impl/src/main/kotlin/net/corda/membership/impl/client/MemberResourceClientImpl.kt

```
@@ -274,15 +276,21 @@ class MemberResourceClientImpl @Activate constructor(
                             KeyValuePairList(emptyList())
                         ),
                     )
-                ).getOrThrow()
-                return RegistrationRequestProgressDto(
-                    requestId,
-                    sent,
-                    SubmittedRegistrationStatus.SUBMITTED,
-                    true,
-                    "Submitting registration request was successful.",
-                    MemberInfoSubmittedDto(memberRegistrationRequest.context)
                 )
+                when (val result = persistentOperation.execute()) {
+                    is MembershipPersistenceResult.Failure -> {
+                        asyncPublisher.publish(persistentOperation.createAsyncCommands().toList())
+                        throw CordaRuntimeException(result.errorMsg)
```

is there a reason we fail here, instead of returning a `SUBMITTED` response with availableNow = false, similar to the catch below?

---

The exception will be caught print the warning and return `availableNow = false`. I wanted not to duplicate the code that prints the warning and create the unavailable reply

---

right that makes sense, I forgot about the catch below